Build failure on runtime-community after #94590

https://dev.azure.com/dnceng-public/public/_build/results?buildId=477820&view=results

This started happening after https://github.com/dotnet/runtime/pull/94590 was merged.

The s390x build fails like this:

> /__w/1/s/src/installer/tests/Assets/Projects/HelloWorld/SelfContained.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.Mono.linux-s390x. No packages exist with this id in source(s): dotnet-eng, dotnet-libraries, dotnet-libraries-transport, dotnet-public, dotnet-tools, dotnet8, dotnet8-transport, dotnet9, dotnet9-transport [/__w/1/s/Build.proj]

While ppc64 and armv6 fail like this:

> /__w/1/s/.dotnet/sdk/8.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(154,5): error NETSDK1084: There is no application host available for the specified RuntimeIdentifier 'linux-armv6'. [/__w/1/s/Build.proj]

/cc @elinor-fung @vitek-karas

--- a/src/installer/tests/Assets/Projects/HelloWorld/SelfContained.csproj
+++ b/src/installer/tests/Assets/Projects/HelloWorld/SelfContained.csproj
@@ -8,6 +8,8 @@
     <UseLocalTargetingRuntimePack>false</UseLocalTargetingRuntimePack>
 
     <!-- Set the RID when building on a platform where we support self-contained + apphost -->
-    <RuntimeIdentifier Condition="'$(TargetsLinux)' == 'true' or '$(TargetsOSX)' == 'true' or '$(TargetsWindows)' == 'true'">$(OutputRID)</RuntimeIdentifier>
+    <_SupportedPlatform Condition="'$(TargetsLinux)' == 'true' or '$(TargetsOSX)' == 'true' or '$(TargetsWindows)' == 'true'">true</_SupportedPlatform>
+    <_SupportedArchitecture Condition="'$(TargetArchitecture)' == 'x64' or '$(TargetArchitecture)' == 'x86' or '$(TargetArchitecture)' == 'arm' or'$(TargetArchitecture)' == 'arm64'">true</_SupportedArchitecture>
+    <RuntimeIdentifier Condition="'$(_SupportedPlatform)' == 'true' and '$(_SupportedArchitecture)' == 'true'">$(OutputRID)</RuntimeIdentifier>
   </PropertyGroup>
 </Project>

